pkg/trace/stats/aggregation.go

```diff
@@ -80,15 +88,54 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 			Synthetics: synthetics,
 		},
 	}
-	if enablePeerSvcAgg {
-		agg.PeerService = s.Meta[tagPeerService]
+	var peerTags []string
+	if spanKindIsClientOrProducer(agg.SpanKind) {
+		if enablePeerSvcAgg {
+			agg.PeerService = s.Meta[tagPeerService]
+		}
+		peerTags = getMatchingPeerTags(s, peerTagKeys)
+		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
-	return agg
+	return agg, peerTags
+}
+
+func getMatchingPeerTags(s *pb.Span, peerTagKeys []string) []string {
+	if len(peerTagKeys) == 0 {
+		return nil
+	}
+	var pt []string
+	for _, t := range peerTagKeys {
+		v, ok := s.Meta[t]
+		if ok {
+			pt = append(pt, t+":"+v)
+		}
+	}
+	if len(pt) == 0 {
+		return nil
+	}
+	return pt
+}
+
+func peerTagsHash(tags []string) uint64 {
+	if len(tags) == 0 {
+		return 0
+	}
+	if !sort.StringsAreSorted(tags) {
+		sort.Strings(tags)
+	}
+	h := fnv.New64a()
+	for i, t := range tags {
+		h.Write([]byte(t))
+		if i != len(tags)-1 {
+			h.Write([]byte{0})
+		}
+	}
```

Comment:
Could you just use `strings.Join` and then do the hash of that, instead? Might simplify this a bit
(https://pkg.go.dev/strings#Join)

Reply:
Discussed offline; the use of `strings.Join` results in 2 extra allocs.